container/openejb-core/src/main/java/org/apache/openejb/assembler/classic/Assembler.java

```diff
@@ -1029,6 +1030,14 @@ private AppContext createApplication(final AppInfo appInfo, ClassLoader classLoa
 
                 systemInstance.fireEvent(new AssemblerAfterApplicationCreated(appInfo, appContext, allDeployments));
                 logger.info("createApplication.success", appInfo.path);
+                
+                //TODO: set proper default
+                //required by spec: EE.5.3.4, used together with tomcat#jndiExceptionOnFailedWrite 
+                if("true".equals(SystemInstance.get().getProperty("forceReadOnlyAppNamingContext", "true"))) {
```

Comment:
please prefix it openejb. (openejb.forceReadOnlyAppNamingContext)

also shouldnt the default be false to ensure it is writable by default (backward compatibility)

also think createApplication.success is not the right key yet ;)